core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -51,15 +63,63 @@ class NewShinyConsumer(topic: Option[String], whitelist: Option[String], consume
   import scala.collection.JavaConversions._
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
-  if (topic.isDefined)
-    consumer.subscribe(List(topic.get))
-  else if (whitelist.isDefined)
-    consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
-  else
-    throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
-
+  consumerInit()
   var recordIter = consumer.poll(0).iterator
 
+  def consumerInit() {
+    if (topic.isDefined)
+      consumer.subscribe(List(topic.get))
+    else if (whitelist.isDefined)
+      consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
+    else
+      throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
+  }
+
+  override def receive(): BaseConsumerRecord = {
+    if (!recordIter.hasNext) {
+      recordIter = consumer.poll(timeoutMs).iterator
+      if (!recordIter.hasNext)
+        throw new ConsumerTimeoutException
+    }
+
+    val record = recordIter.next
+    BaseConsumerRecord(record.topic,
+                       record.partition,
+                       record.offset,
+                       record.timestamp,
+                       record.timestampType,
+                       record.key,
+                       record.value)
+  }
+
+  override def stop() {
+    this.consumer.wakeup()
+  }
+
+  override def cleanup() {
+    this.consumer.close()
+  }
+
+  override def commit() {
+    this.consumer.commitSync()
+  }
+}
+
+class NewShinySimpleConsumer(topic: String, partitionId: Int, offset: Long, consumerProps: Properties, timeoutMs: Long = Long.MaxValue) extends NewShinyConsumer(Some(topic), None, consumerProps, timeoutMs) {
+  import org.apache.kafka.clients.consumer.KafkaConsumer
+
+  import scala.collection.JavaConversions._
+
+  override def consumerInit() {
+    val topicPartition = new TopicPartition(topic, partitionId)
+    consumer.assign(List(topicPartition))
+    offset match {
+      case OffsetRequest.EarliestTime => consumer.seekToBeginning(List(topicPartition))
+      case OffsetRequest.LatestTime => consumer.seekToEnd(List(topicPartition))
+      case _ => consumer.seek(topicPartition, offset)
+    }
+  }
+
```

Comment:
The methods after this line in this class seem to be unnecessary duplicates (they were part of `NewShinyConsumer` and they got copied between the two). We can get rid of the ones in `NewShinySimpleConsumer` since it inherits from `NewShinyConsumer`.